akka-actor/src/main/scala/akka/actor/FSM.scala

```diff
@@ -141,6 +145,7 @@ object FSM {
      */
     def forMax(timeout: Duration): State[S, D] = timeout match {
       case f: FiniteDuration ⇒ copy(timeout = Some(f))
+      case Duration.Inf      ⇒ copy(timeout = SomeMaxFiniteDuration)
       case _                 ⇒ copy(timeout = None)
```

Comment:
RE: https://twitter.com/Gentmen/status/586192587654041600
The other mentioned values are not classes but values in `Duration`, as seen here:
https://github.com/scala/scala/blob/2.11.x/src/library/scala/concurrent/duration/Duration.scala#L174